packages/devtools_app/lib/src/screens/memory/memory_charts.dart

```diff
@@ -155,14 +153,16 @@ class ChartsValues {
 
   bool get hasManualGc => _event.containsKey(manualGCJsonName);
 
-  bool get hasGc => _vm[gcJsonName];
+  bool get hasGc => _vm[gcJsonName] as bool;
 
   int get extensionEventsLength =>
       hasExtensionEvents ? extensionEvents.length : 0;
 
   List<Map<String, Object>> get extensionEvents {
     if (_extensionEvents.isEmpty) {
-      _extensionEvents.addAll(_event[extensionEventsJsonName]);
+      _extensionEvents.add(<String, Object>{
```

Comment:
why did we have to move away from `addAll`? The result of this call is not equivalent to what we had before:

Let: 
```
_extensionEvents = {'key1': 'value1'};
_event[extensionEventsJsonName] = {'key2': 'value2'};
```
Before:
```
_extensionEvents.addAll(_event[extensionEventsJsonName]);
--->
      {
          'key1': 'value1',
          'key2': 'value2',
      }
```

After: 
```
_extensionEvents.add({
    extensionEventsJsonName: _event[extensionEventsJsonName]!,
});
--->
      {
          'key1': 'value1',
          extensionEventsJsonName: {'key2': 'value2'},
      }
```

Reply:
fixed